tests/raftstore/test_transfer_leader.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::util::*;
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+
+fn test_transfer_leader<T: Simulator>(cluster: &mut Cluster<T>) {
+    // test a cluster with five nodes [1, 5], only one region (region 1).
+    // every node has a store and a peer with same id as node's.
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    // transfer leader to (2, 2)
+    cluster.transfer_leader(1, new_peer(2, 2));
```

Comment:
add a test after transfer_leader but still send command to peer (1, 1) directly and check whether we can execute this command.